cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/frontends/Language.kt

```diff
@@ -361,6 +364,29 @@ abstract class Language<T : LanguageFrontend<*, *>> : Node() {
             Pair(list.toSet(), CallResolutionResult.SuccessKind.SUCCESSFUL)
         }
     }
+
+    /**
+     * This function returns the best viable declaration when resolving a [Reference]. The
+     * candidates to chose from are stored in [Reference.candidates] In most cases the languages can
+     * keep the default implementation, which only returns a declaration, if the list contains one
+     * single item. Otherwise, we have an ambiguous result and cannot determine the result with
+     * certainty.
+     *
+     * If we encounter an ambiguous result, a warning is issued.
+     */
+    open fun bestViableReferenceCandidate(ref: Reference): Declaration? {
+        return if (ref.candidates.size > 1) {
+            Util.warnWithFileLocation(
+                ref,
+                log,
+                "Resolution of reference {} was ambiguous, cannot set refersTo correctly.",
```

Comment:
I would mention what is used when this warning is printed. Here `null`.

Reply:
What do you mean exactly?